Document how to recover objects from backup (#55)
appuio/openshift4-docs: docs/modules/ROOT/pages/how-tos/recover-from-backup.adoc

PASSWORD_KEY="$(cat catalog/manifests/cluster-backup/* | yq read -cjd'*' - | jq -r '.[] | select(.kind == "Secret" and .metadata.name == "objects-backup-password") | .stringData.password' | cut -d: -f2)"
export RESTIC_PASSWORD=$(vault kv get -format json "clusters/kv/${PASSWORD_KEY%/*}" | jq -r ".data.data.${PASSWORD_KEY##*/}")

ID_KEY="$(cat catalog/manifests/cluster-backup/* | yq read -cjd'*' - | jq -r '.[] | select(.kind == "Secret" and .metadata.name == "object-backup-s3-credentials") | .stringData.username' | cut -d: -f2)"
export AWS_ACCESS_KEY_ID=$(vault kv get -format json "clusters/kv/${ID_KEY%/*}" | jq -r ".data.data.${ID_KEY##*/}")

SECRET_KEY="$(cat catalog/manifests/cluster-backup/* | yq read -cjd'*' - | jq -r '.[] | select(.kind == "Secret" and .metadata.name == "object-backup-s3-credentials") | .stringData.password' | cut -d: -f2)"
export AWS_SECRET_ACCESS_KEY=$(vault kv get -format json "clusters/kv/${SECRET_KEY%/*}" | jq -r ".data.data.${SECRET_KEY##*/}")
----

== Identify and retrieve snapshot

1. List the available snapshots.
   Identify the one you do want to restore.

>    Take note its `ID`.

Reviewer: ```suggestion
   Take note of its `ID`.
```

+
[source,console]
----
restic snapshots
----

2. Retrieve the backup archive
+
[source,console]
----
restic restore <ID> --target . 
----

== Extract and prepare files

1. List files in the backup.
   Take note of the path containing the required files.
+
[source,console]
----
tar tvf syn-cluster-backup-object-dumper.tar.gz
----

> 2. Extract required files

Reviewer: ```suggestion
2. Extract required files.
```

   If all files should be extracted, `path/inside/archive` can be omitted.
   Files will be put in the directory `restore` within the current working directory.
+
[source,console]
----
mkdir restore
tar -C restore -xf syn-cluster-backup-object-dumper.tar.gz [path/inside/archive]
----

3. Prepare files
+

> Depending on the restores requirement, the extracted files need to be altered before they can be applied to the cluster.

Reviewer: this looks a bit strange
```suggestion
Depending on the restore requirements, the extracted files need to be altered before they can be applied to the cluster.
```
